### ISSUE TITLE
[PS][D8] Replace uses of drupal_get_path() with dirname(__FILE__) where possible

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4684]
This is the respective issue as https://www.drupal.org/project/drupal/issues/2351919 which was merged in D8 core, and is slated for backport to D7 core:

> There are many uses of `drupal_get_path()` in where the simple use of `__DIR__` would have been sufficient and more performant.
>
> ## Proposed resolution
Replace occurrences of `drupal_get_path()` with `__DIR__` wherever the targeted file is known to share the same module directory with the calling code (and whose relative position will not change dynamically), and thus can be hard-coded.

...

> This could be considered for Drupal 7 backport (though with `dirname(__FILE__)` rather than `__DIR__` for PHP version compatibility reasons).
>
> I think it would mainly just be a micro-optimization that avoids some extra function calls, since `drupal_get_path()` itself should already be incredibly fast (see https://www.drupal.org/node/2442383#comment-10827200 for why) so I don't know if it's worth doing, but maybe it still is.

**Comment by domaingood** [comment]
+1

**Comment by indigoxela** [comment]
At least the issue title should mention the actual function name **backdrop**_get_path(). :wink: 

`dirname(__FILE__)` isn't necessary, because __DIR__ came with php 5.3.0 and Backdrop isn't supposed to support even older versions.

**Comment by klonos** [comment]
He he 😄

**Comment by klonos** [comment]
> `dirname(__FILE__)` isn't necessary, because `__DIR__` came with php 5.3.0 ...

That's true: https://www.php.net/manual/en/language.constants.predefined.php. We currently officially support 5.3.2+ (see https://backdropcms.org/requirements).

**Comment by alanmels** [comment]
I am also all for this kind of improvements which positively effect performance.

**Comment by indigoxela** [comment]
Whoa, that's a lot...

Here's a quick bash one-liner. Note, it very likely still contains false positives.
```
cd core/modules
ls -1 | while read LINE; do grep -r "backdrop_get_path('module', '$LINE')" $LINE; done | wc -l

194
```

**Comment by BWPanda** [comment]
> Replace occurrences of drupal_get_path() with __DIR__ **wherever the targeted file is known to share the same module directory with the calling code (and whose relative position will not change dynamically)**, and thus can be hard-coded.

That seems a bit complicated to figure out. To me, backdrop_get_path() is easy to understand. Also:

> since drupal_get_path() itself should already be incredibly fast ... so I don't know if it's worth doing

Do we know how much faster __DIR__ would be (if at all) in Backdrop? Otherwise I don't see the need...
